docs/design/value-attribution.md

```diff
@@ -0,0 +1,79 @@
+# Value Attribution
+
+### Brandon Iglesias, Dennis Coyle
+
+
+## Definitions
+
+*Attribution* - _Acknowledgment for establishing the relationship between Storj and a customer_
+
+## OVERVIEW
+
+When our partners bring data to the Tardigrade network we want the ability to attach attribution for the data. 
+
+Our partners will have connectors that their customers will use to store data on the Tardigrade network. When one of their customers uses the connector to store data in a bucket on the Tardigrade network, we want to give that partner attribution for the data that is stored in it.
+
+## GOALS
+
+
+* A generic library that can be used in all  Golang third party connectors
+
+* Data attribution on a per bucket basis
+
+* Ability to calculate how much data was stored in the bucket for a given amount of time. 
+
+* Ability to calculate attributed data bandwidth on the Storj Network
+
+
+## NON GOALS
+
+* Attributions per object
+
+* Multiple attributions per bucket
+
+* Attributions per project
+
+* Attributions per satellite account
+
+
+
+## SCENARIOS
+
+
+* A client points partner connector A to a bucket with NO data in it.
+
+	* The partner receives the attribution for all of the data that the client stores and retrieves from it. 
+
+	* The bucket must be empty when the partner connector is connected to it. 
+
+* A client points partner connector A to an existing bucket that NO OTHER connector has pointed to but has data in it. 
+
+	* The partner does NOT receive attribution for data stored or retrieved from that bucket. 
+
+* A client points partner connector A to an existing bucket that partner connector B had already been pointed to. 
+
+	* Partner connector B receives the attribution for all of the data that is stored and retrieved from the bucket. 
+
+## DESIGN
+### Connector
+Each partner will have a registered uuid that will identify a partners connector on the Storj network. Each connector will implement the S3 client interface to allow operations through the connector.  When a user uploads data to a specified bucket through the connector,  the connector will include the partner uuid in the context of the request. Before an upload occurs, the uplink will communicate the uuid and bucket name with the tardigrade satellite, checking for a previous attribution. If no attribution is found on the specified bucket and the bucket is currently void of data, the satellite will attribute the partners uuid to that bucket within the metadata struct.
```

Comment:
So Value Attribution will be limited only to S3 client interface?

Reply:
a few thoughts:
1) we probably don't need a full UUID. anything clearly identifying the connector is fine. The way GCP and Azure work is they use the User Agent string. (Here is Minio setting the User Agent for Azure: https://github.com/minio/minio/blob/master/cmd/gateway/azure/gateway-azure.go#L184, where Minio plays the part of connector and Azure is us).
2) connector attribution should work for anything that uses libuplink of any kind - so there isn't necessarily an S3 interface. But libuplink should allow for setting some sort of identifier (UUID, user agent) before performing operations on a satellite.

Reply:
Was an extraneous sentence explaining how connectors will be built. My assumption they would implement the S3 interface was wrong but the way this works doesn't require S3 compatibility. 
`the connector will include the partner uuid in the context of the request`.